core/src/main/java/com/digitalpebble/storm/crawler/protocol/httpclient/HttpProtocol.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.digitalpebble.storm.crawler.protocol.httpclient;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.Header;
+import org.apache.http.HeaderIterator;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.DefaultProxyRoutePlanner;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.LoggerFactory;
+
+import backtype.storm.Config;
+
+import com.digitalpebble.storm.crawler.Metadata;
+import com.digitalpebble.storm.crawler.protocol.AbstractHttpProtocol;
+import com.digitalpebble.storm.crawler.protocol.HttpRobotRulesParser;
+import com.digitalpebble.storm.crawler.protocol.ProtocolResponse;
+import com.digitalpebble.storm.crawler.protocol.RobotRulesParser;
+import com.digitalpebble.storm.crawler.util.ConfUtils;
+
+import crawlercommons.robots.BaseRobotRules;
+
+/**
+ * Uses Apache httpclient to handle http and https
+ **/
+
+public class HttpProtocol extends AbstractHttpProtocol implements
+        ResponseHandler<ProtocolResponse> {
+
+    private static final org.slf4j.Logger LOG = LoggerFactory
+            .getLogger(HttpProtocol.class);
+
+    private final static PoolingHttpClientConnectionManager CONNECTION_MANAGER = new PoolingHttpClientConnectionManager();
+    static {
+        // Increase max total connection to 200
+        CONNECTION_MANAGER.setMaxTotal(200);
+        // Increase default max connection per route to 20
+        CONNECTION_MANAGER.setDefaultMaxPerRoute(20);
```

Comment:
sure, bit of a pain as we're static here and the conf would come from configure() which is not. Let's not aim for perfection straight away and do that in the next version ;-) the default values should be fine for most cases